### PR TITLE
add jest setup file and the component snapshot test

### PR DESCRIPTION
**Pull request by libangrui**

### CODE REVIEW

File: tests/components/Base/Button.test.js [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
 
-  it('primary type', ()=> {
-    const wrapper=shallow(<Button className='primary' />);
-    expect(wrapper.hasClass('primary')).toBeTruthy();
+    expect(toJson(wrapper)).toMatchSnapshot();
   });
 
   it('call onClick', ()=> {
-    //
+    const mockClick = jest.fn();
```
Comment (iwisunny) [inline tests/components/Base/Button.test.js]: why on node repl, `jest.fn` is undefined ?

File: tests/setup.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+global.window.resizeTo = (width, height) => {
```
[the remark this answers is not in the document]
Comment (libangrui) [reply tests/setup.js]: It is not used at this time. I will delete it for now.

File: tests/components/Base/Popover.test.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { render, mount } from 'enzyme';
```
[the remark this answers is not in the document]
Comment (libangrui) [reply tests/components/Base/Popover.test.js]: ok

File: tests/components/Base/Table.test.js [hunk 4]
```diff
@@ -1,19 +1,10 @@
...
-
-const setup = (props = {}, type) => {
-  const wrapper = type === 'mount' ? mount(<Table {...props}/>) : shallow(<Table {...props}/>);
-
-  return {
-    wrapper,
-    props: wrapper.instance().props,
-  };
-};
+const setup = (props = {}, type) => (type === 'render' ? render(<Table {...props} />) : mount(<Table {...props} />));
```
[the remark this answers is not in the document]
Comment (libangrui) [reply tests/components/Base/Table.test.js]: ok, i adjust it.

**Pull request merged**